mimesniff.bs

```
@@ -139,553 +161,283 @@ Indent: 1
+<a spec=http>media-type</a> token production. In particular, a <a>valid MIME type</a> may include
+<a for="MIME type">parameters</a>. [[!RFC7231]]
 
-<h3 id=parsing-a-mime-type>Parsing a MIME type</h3>
+<p class=note>A <a>valid MIME type string</a> is supposed to be used for conformance checkers only.
 
-<p>
- To <dfn>parse a MIME type</dfn>, the user agent must execute the following
- steps:
+<div class=example id=example-valid-mime-type-string>
+ <p>"<code>text/html</code>" is a <a>valid MIME type string</a>.
 
- <ol>
-  <li>
-   Let <var>sequence</var> be the <a>byte sequence</a> of the <a>MIME
-   type</a>, where <var>sequence</var>[<var>s</var>] is <a>byte</a>
-   <var>s</var> in <var>sequence</var> and <var>sequence</var>[0] is the first
-   <a>byte</a> in <var>sequence</var>.
+ <p>"<code>text/html;</code>" is not a <a>valid MIME type string</a>, though
+ <a>parse a MIME type</a> returns a <a>MIME type record</a> for it identical to if the input had
+ been "<code>text/html</code>".
+</div>
 
-  <li>
-   If the number of <a>bytes</a> in <var>sequence</var> is
-   less than 1, return undefined.
+<p>A <dfn export>valid MIME type string with no parameters</dfn> is a <a>valid MIME type string</a>
+that does not contain U+003B (;).
 
-  <li>
-   Initialize <var>s</var> to 0.
 
-  <li>
-   Initialize <var>type</var> and <var>subtype</var> to the empty string
-   ("<code></code>").
+<h3 id=parsing-a-mime-type>Parsing a MIME type</h3>
 
-  <li>
-   Initialize <var>parameters</var> to the empty dictionary ({}).
+<p>To <dfn export>parse a MIME type</dfn>, given a string <var>input</var>, run these steps:
 
-  <li>
-   While <var>sequence</var>[<var>s</var>] is <a>ASCII whitespace</a>,
-   continuously execute the following steps:
+<ol>
+ <li><p>Remove any leading and trailing <a>ASCII whitespace</a> from <var>input</var>.
 
-   <ol>
-    <li>
-     Increment <var>s</var> by 1.
+ <li><p>If <var>input</var> is the empty string, then null.
```

return null.

The algorithm's intro should probably mention that null is the failure condition. (Or, use "failure" instead of "null".)

---

It's not a failure condition.

---

Then what MIME type gets produced?

---

I ended up removing this. It wasn't needed after all. Initially it was meant to let callers handle that case in a special way, but further inspection revealed that not to be necessary. Not sure why I initially thought it was.